src/libopensc/pkcs15-sec.c

```diff
@@ -448,7 +448,7 @@ int sc_pkcs15_wrap(struct sc_pkcs15_card *p15card,
 		const struct sc_pkcs15_object *key,
 		struct sc_pkcs15_object *target_key,
 		unsigned long flags,
-		u8 * cryptogram, size_t* crgram_len,
+		u8 * cryptogram, unsigned long* crgram_len,
```

Comment:
This seems bogus change, size_t should be "semantically correct" here?

Reply:
If I remember correctly, the reason for this is that size of size_t varies on different architectures but where this function is called (in framework-pkcs15) this argument is casted from PKCS#11 type CK_ULONG* and CK_ULONG is defined to be at least 32 bits. So if for example size_t is 16 bits (could be on some embedded architectures) and CK_ULONG is 32 bits, this causes a compilation error. While I also thought that size_t* looks better I ended up putting it this way to avoid type conversions in framework-pkcs15.c and also to be consistent with sc_pkcs15_derive where *poutlen is an unsigned long pointer too.